### DISCUSSION TITLE
Simulating the consolidation (compaction) after drainage

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#17913] [Q&A Modules: Solid mechanics] [answered]
Hi guys

I want to simulate the consolidation (compaction) of the porous soil after the single-phase water is drained out from the system, i'd like to see how much will the geometry shrink (disp_z) with time.

I know that porous flow module can be coupled with tensor-mechanics for this kind of simulation (PorousFlowBasicTHM).
Is there any example of the drainage simulation that I can learn from?

Thanks for your help.

Kind regards,
Traiwit

**Comment by WilkAndy** [comment, marked as answer]
Yes, have a look at

https://mooseframework.inl.gov/modules/porous_flow/tests/poro_elasticity/poro_elasticity_tests.html

I strongly suggest that you use one of the PorousFlowActions, either https://mooseframework.inl.gov/source/actions/PorousFlowBasicTHM.html or https://mooseframework.inl.gov/source/actions/PorousFlowFullySaturated.html , depending on your problem, as they will simplify your input file and you won't have to think about user-unfriendly things like numerical stabilization.  The reason most of the "poroelasticity tests" don't use the Actions is

- for exact comparison to analytic solutions very specific Kernels and Materials have to be used - hence the long, boring discussion of these in the poroelastic doco, above (the Mandel problem, which is probably very similar to what you want has 4 versions, sigh, with only the last one using the Action)
- i wanted to illustrate how different choices can slightly impact the results

But when you're just building a "real" model (not comparison with analytic solutions) then an Action will be much easier.

**Reply by Traiwit** [reply]
Thanks @WilkAndy !,

I will start learning the Mandel example (FullSatVol).
As I see, the example drains the fluid out by squeezing, I just need to figuere out the way to move the fluid out from the system by gravity only.

Kind regards,
Traiwit

**Reply by WilkAndy** [reply]
yep, just apply appropriate BCs (eg, zero porepressure, or hydraulic-head gradient)
